tests/libraries/token/test_standard_token.py

```diff
@@ -36,4 +49,80 @@ def test_eternal_approval_magic(contractsFixture):
     assert standardToken.allowance(tester.a0, tester.a1) == 2 ** 256 - 1
     assert standardToken.balanceOf(tester.a0) == 4
     assert standardToken.balanceOf(tester.a1) == 96
-    
\ No newline at end of file
+    
+def test_create_negative_balance(testerContractsFixture):
+    standardToken = testerContractsFixture.contracts['StandardTokenHelper']
+    assert standardToken.totalSupply() == 0
+    # We get some tokens for tester 0
+    assert standardToken.faucet(100)
+    assert standardToken.balanceOf(tester.a0) == 100
+    # Tester 1 tries to send tokens from tester 0 to himself and fails
+    with raises(TransactionFailed):
+        standardToken.transferFrom(tester.a0, tester.a1, 101)
+
+def test_transfer_then_create_negative_balance(testerContractsFixture):
+    standardToken = testerContractsFixture.contracts['StandardTokenHelper']
+    assert standardToken.totalSupply() == 0
+    # We get some tokens for tester 0
+    assert standardToken.faucet(10)
+    assert standardToken.balanceOf(tester.a0) == 10
+    # approve and transfer 10 to other tester
+    assert standardToken.approve(tester.a1, 10)
+    assert standardToken.allowance(tester.a0, tester.a1) == 10
+    assert standardToken.transfer(tester.a1, 10)
+    assert standardToken.balanceOf(tester.a1) == 10
+    # approve and allow transfer of more tokens than tester has
+    assert standardToken.approve(tester.a1, 11)
+    # try to transfer more than tester has  
+    with raises(TransactionFailed):
+        standardToken.transfer(tester.a1, 11)
```

Comment:
I don't believe this test is doing what you want.  You start out by giving 10 tokens to `a0`.  You then approve `a1` to spend your 10 tokens.  Then you transfer 10 tokens to `a1`, but the approval isn't used at all because the approval was to allow `a1` to transfer the tokens but you transferred the tokens as `a0` (the token owner).  If you want `a1` to transfer on behalf of `a0` then you need to use `transferFrom(tester.a0, tester.a2, 10, sender=tester.k1)`.  This will have `a1` send 10 of `a0`s tokens to `a2`.

Reply:
Alternatively, just remove the approvals if your intent is to test the `transfer` function rather than `approvals`.

Reply:
Same issue in a couple other tests where you are setting an approval and then not leveraging it at all in the test.